dm/master/server.go

```diff
@@ -360,15 +362,27 @@ func (s *Server) StartTask(ctx context.Context, req *pb.StartTaskRequest) (*pb.S
 	if len(sourceRespCh) > 0 {
 		sourceResps = sortCommonWorkerResults(sourceRespCh)
 	} else {
+		sources := make([]string, 0, len(stCfgs))
+		for _, stCfg := range stCfgs {
+			sources = append(sources, stCfg.SourceID)
+		}
+		if req.RemoveMeta {
+			if scm := s.scheduler.GetSubTaskCfgsByTask(cfg.Name); len(scm) > 0 {
```

Comment:
Do we need any mechanism to prevent `start-task` in another client when removing meta?

Reply:
Add `removeMetaLock`

Reply:
why need a lock, may start task with same task name?

Reply:
em... may user `start-task` again when the previous `start-task` is still removing meta?